src/main/java/com/reshiftsecurity/analytics/ReshiftAnalyticsService.java

```diff
@@ -0,0 +1,32 @@
+package com.reshiftsecurity.analytics;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ReshiftAnalyticsService {
+    List<ReshiftAnalyticsAction> actions;
+    Gson jsonSerializer;
+
+    public ReshiftAnalyticsService() {
+        this.actions = new ArrayList<>();
+        this.jsonSerializer = new GsonBuilder().setPrettyPrinting().create();
+    }
+
+    public void recordAction(ReshiftAnalyticsAction action) {
+        this.actions.add(action);
+        this.processActions();
+    }
+
+    private void processActions() {
+        if (actions.size() >= 100) {
```

Comment:
what's this magic number for?

Reply:
Haven't decided on that yet, this code is not yet utilised. The whole analytics piece still needs design work.

Reply:
To answer your question better: essentially we need to decide on how we can gather analytics. Most likely it will involve hitting some API. Initially I thought we should have a number of entries in before we call the api. But stopped there so I committed the code as is. So the number should be configurable as well.

Reply:
Ticket for addressing analytics: https://softwaresecured.atlassian.net/browse/OM-1341